core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -5269,6 +5274,20 @@ private static SqlNode getNthExpr(SqlNode query, int ordinal, int sourceCount) {
     SqlInsert insertCallAfterValidate = call.getInsertCall();
     if (insertCallAfterValidate != null) {
       validateInsert(insertCallAfterValidate);
+      // Validate NULL
```

Comment:
I didn't understand the 'Validate NULL' comment at first. 'Throw if select list contains NULL literal and target is NOT NULL' might be better.

Reply:
Good idea.